src/main/java/com/direwolf20/buildinggadgets/common/building/Region.java

```diff
@@ -0,0 +1,345 @@
+package com.direwolf20.buildinggadgets.common.building;
+
+import com.direwolf20.buildinggadgets.common.tools.GadgetUtils;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.PeekingIterator;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3i;
+
+import java.util.Objects;
+import java.util.Spliterator;
+
+/**
+ * Represents a region in the world with a finite and nonzero size.
+ * <p>Javadoc are copied from {@link AxisAlignedBB} with some modifications.</p>
+ */
+public final class Region implements IPlacementSequence {
+
+    private final int minX;
+    private final int minY;
+    private final int minZ;
+    private final int maxX;
+    private final int maxY;
+    private final int maxZ;
+
+    public Region(int minX, int minY, int minZ, int maxX, int maxY, int maxZ) {
+        this.minX = Math.min(minX, maxX);
+        this.minY = Math.min(minY, maxY);
+        this.minZ = Math.min(minZ, maxZ);
+        this.maxX = Math.max(minX, maxX);
+        this.maxY = Math.max(minY, maxY);
+        this.maxZ = Math.max(minZ, maxZ);
+    }
+
+    public Region(Vec3i vertex) {
+        this(vertex, vertex);
+    }
+
+    public Region(Vec3i min, Vec3i max) {
+        this(min.getX(), min.getY(), min.getZ(), max.getX(), max.getY(), max.getZ());
+    }
+
+    /**
+     * Translates this Region by the given amount
+     *
+     * @param x how much to translate x
+     * @param y how much to translate y
+     * @param z how much to translate z
+     * @return A new Region, translated by the given coordinates
+     */
+    public Region translate(int x, int y, int z) {
+        return new Region(minX + x, minY + y, minZ + z, maxX + x, maxY + y, maxZ + z);
+    }
+
+    /**
+     * @see #translate(int, int, int)
+     */
+    public Region translate(Vec3i direction) {
+        return this.translate(direction.getX(), direction.getY(), direction.getZ());
+    }
+
+    /**
+     * @param x x-growth
+     * @param y y-growth
+     * @param z z-growth
+     * @return A new Region who's max coordinates are increased by the given amount
+     */
+    public Region grow(int x, int y, int z) {
+        return new Region(minX, minY, minZ, maxX + x, maxY + y, maxZ + z);
+    }
+
+    /**
+     * See {@link #grow(int, int, int)} - grow on all three axises.
```

Comment:
```suggestion
     * See {@link #grow(int, int, int)} - grown on all three axises.
```

Reply:
Not so sure about this, suggestion:
`See {@link #grow(int, int, int)} - grows on all three axes.`